### DISCUSSION TITLE
Idea for v2: multiple, federated deployments

### DISCUSSION THREAD

**Discussion by shalak** [gameyfin/gameyfin#236] [Feature Requests]
The usecase I have in mind is to have two gameyfin instances:
- main one, that I use myself and within family
- "guest" one, that will be hosted of different machine (a portable LAN party server)

The idea is, that when I'm hosting a LAN party at my house, the guests are connecting to the guest instance, and it can serve a game for download either from my main instance, or from local copy (if copy is present locally). The local copy is populated either manually by admin, or on first demand to download by a user.

When I pick up this "roadwarrior" setup, and take the server to some other place, outside of my home network, I'm still able to serve the games from local copies.

**Comment by grimsi** [comment]
I don't think that there is anything that has to be implemented in Gameyfin here since it could be solved using the existing features:

1. Create a Gameyfin instance (the "main" one) with two libraries ("personal" and "guest")
2. Create a second Gameyfin instance (the "guest" one)
3. Use tools like rsync to sync the "guest" library between these two instances

Regarding your wish to configure the "guest" instance as kind of a read-through cache: This seems like a super niche use-case with a huge impact on development. I want to avoid feature-creep at all costs. So currently this will not be something that I consider useful for Gameyfin.
